src/main/java/Duke.java

```diff
@@ -162,29 +187,46 @@ public static void addToList(Duke duke, String taskdescription) {
         } else {
             task = new ToDo(taskdescription);
         }
+        FileWriter fw = new FileWriter(duke.taskFile, true);
+        fw.write(task.printTask() + "\n");
+        fw.close();
         System.out.println("       " + task.printTask());
         duke.listToDo.add(task);
         System.out.format("     Now you have %d tasks in the list.\n", duke.listToDo.size());
         System.out.println("    ____________________________________________________________\n");
     }
 
-    public static void markTaskDone(Duke duke, int index) {
+    public static void markTaskDone(Duke duke, int index) throws IOException {
         System.out.println("    ____________________________________________________________");
         Task task = duke.listToDo.get(index - 1);
         task.markAsDone();
+        FileWriter fw = new FileWriter(duke.taskFile);
+        String toWrite = "";
+        for (Task task1 : duke.listToDo) {
+            toWrite += task1.printTask() + "\n";
+        }
+        fw.write(toWrite);
+        fw.close();
         System.out.println("     Nice! I've marked this task as done: ");
         System.out.println("       " + task.printTask());
         System.out.println("    ____________________________________________________________\n");
     }
 
-    public static void deleteTask(Duke duke, int index) {
+    public static void deleteTask(Duke duke, int index) throws IOException {
         System.out.println("    ____________________________________________________________");
         Task task = duke.listToDo.get(index - 1);
         System.out.println("     Noted. I've removed this task: ");
```

Comment:
May I ask for the purpose of the spaces in front of your output string? Is this for aesthetic purposes? I have noticed this in a few other places as well.